### PR TITLE
Speed up main view with project main data

### PR DESCRIPTION
**Pull request by rldhont**
Adding a class to get only project main data for main view

### CODE REVIEW

**Review by mdouchin — APPROVED** [review]

File: lizmap/modules/lizmap/lib/Project/ProjectMainData.php [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+
+        return false;
+    }
+
+    /**
+     * Get option hide project in a Lizmap config content.
+     *
+     * @param object $cfgContent - the Lizmap config content
+     *
+     * @return bool
```
Comment (nworr) [inline lizmap/modules/lizmap/lib/Project/ProjectMainData.php:505]:
```suggestion

```

File: lizmap/modules/lizmap/lib/Project/ProjectMainData.php [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+     */
+    protected function getHiddenOption($cfgContent)
```
Comment (nworr) [inline lizmap/modules/lizmap/lib/Project/ProjectMainData.php:507]: We can use explicit return type (at least for simple return type ?) 
 ```suggestion
     protected function getHiddenOption($cfgContent): bool
```

**Review by nworr — APPROVED** [review]
Looks good to me.
I made a suggestion about return type but it's a wider subject.

Thanks , specially for the tests

**Review by nboisteault — APPROVED** [review]

**Comment by mdouchin** [comment]
Profile with XDebug on the index.php (Lizmap welcome page). Visualized with KCacheGrind in "Memory" mode. Complexity decreased

* Current Master

image


* This PR

image

**Pull request merged**